front-end-tools/src/components/DeployComponent.tsx

```diff
@@ -0,0 +1,268 @@
+                                    Buffer.from(sha256([new Uint8Array(arrayBuffer)])).toString('hex')
+                                );
+
+                                // Concordium's tooling create versioned modules e.g. `.wasm.v1` now.
+                                // Unversioned modules `.wasm` cannot be created by Concordium's tooling anymore.
+                                // If the module is versioned, the first 8 bytes are the version, followed by the `magicValue` below.
+                                // If the module is an old unversioned one, the module starts with the `magicValue` below.
+                                const magicValue = new Uint8Array([0x00, 0x61, 0x73, 0x6d]);
+                                let uploadedModuleFirst4Bytes = new Uint8Array([]);
+
+                                if (arrayBuffer.byteLength >= 4) {
+                                    uploadedModuleFirst4Bytes = new Uint8Array(arrayBuffer).subarray(0, 4);
+                                } else {
+                                    setUploadError(`You might have not uploaded a Concordium module.`);
+                                }
+
+                                // If we have an unversioned module, we remove no bytes.
+                                // If we have a versioned module, we remove 8 bytes (remove the versioned 8 bytes at the beginning)
+                                const slice = arraysEqual(uploadedModuleFirst4Bytes, magicValue) ? 0 : 8;
+
+                                let wasmModule;
+                                try {
+                                    wasmModule = await WebAssembly.compile(arrayBuffer.slice(slice));
+                                } catch (err) {
+                                    setUploadError(
+                                        `You might have not uploaded a Concordium module. Original error: ${
+                                            (err as Error).message
+                                        }`
+                                    );
+                                }
+
+                                if (wasmModule) {
+                                    const moduleFunctions = WebAssembly.Module.exports(wasmModule);
+
+                                    const contractNames = [];
+                                    for (let i = 0; i < moduleFunctions.length; i += 1) {
+                                        if (moduleFunctions[i].name.slice(0, 5) === 'init_') {
+                                            contractNames.push(moduleFunctions[i].name.slice(5));
+                                        }
+                                    }
+                                    setContracts(contractNames);
+
+                                    const customSection = WebAssembly.Module.customSections(
+                                        wasmModule,
+                                        'concordium-schema'
```

Comment:
This is the current version. If somebody has an older version there might be `concordium-schema-v1` and `concordium-schema-v2` sections as well which have the "unversioned module schema".

There are probably some modules that are deployed still with the old schema.

Reply:
True, but since Concordium tooling does not produce such old modules anymore and we assume that the `DeployComponent` is used by current smart contract developers who are actively developing and compiling fresh modules, it might be ok to not support very old modules here.

If you feel strongly about it, I suggest we add this as a future task.

Reply:
The `Update` and `Read` step get their embedded module from `rpcClient.getEmbeddedSchema(moduleRef)` which should support old schemas as long as we haven't removed support for it.